addons/mail/static/src/new/thread/message_service.js

```diff
@@ -431,8 +438,12 @@ export class MessageService {
         group.resIds.add(data.resId);
     }
 
+    /**
+     * @param {import("@mail/new/core/thread_model").Thread} thread
+     * @returns {import("@mail/new/core/thread_model").Thread["messages"]}
+     */
     sortMessages(thread) {
-        thread.messages.sort((msgId1, msgId2) => {
+        return thread.messages.sort((msgId1, msgId2) => {
```

Comment:
Don't rely on return. This method sort the thread in place, just inspect `thread.messages` after calling it.